src/test/java/org/example/controller/ApplicationControllerTest.java

```diff
@@ -0,0 +1,139 @@
+package org.example.controller;
+
+import org.example.controllers.ApplicationController;
+import org.example.exceptions.DoesNotExistException;
+import org.example.exceptions.FailedToCreateException;
+import org.example.exceptions.InvalidException;
+import org.example.models.ApplicationRequest;
+import org.example.models.ApplicationResponse;
+import org.example.services.ApplicationService;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import javax.ws.rs.core.Response;
+
+import java.sql.SQLException;
+
+import static org.junit.Assert.assertEquals;
+
+public class ApplicationControllerTest {
+
+    ApplicationService applicationService = Mockito.mock(ApplicationService.class);
+
+    ApplicationController applicationController = new ApplicationController(applicationService);
+
+    ApplicationRequest applicationRequest = new ApplicationRequest(
+            "user@kainos.com",
+            1,
+            "https://academy-job-portal-cvs.s3.eu-west-1.amazonaws.com/the_extra_semicolon/TestDocument.docx"
```

Comment:
It's a mock use fake data, due to this being a public repo you have exposed the S3 link to the world. It's a security vulnerability and you will need to squash your commits.

Reply:
This has been fixed.